transaction/core/src/tx.rs

```diff
@@ -193,25 +192,12 @@ impl TxPrefix {
             .collect()
     }
 
-    /// Get the commitment and blinding for the fee output.
-    pub fn fee_value_and_blinding(&self) -> (u64, Scalar) {
-        (self.fee, Scalar::zero())
-    }
-
-    /// Get all output commitments (including explicit outputs and the implicit fee output).
+    /// Get all output commitments.
     pub fn output_commitments(&self) -> Vec<CompressedCommitment> {
-        let mut commitments: Vec<CompressedCommitment> = self
-            .outputs
+        self.outputs
             .iter()
             .map(|output| output.amount.commitment)
-            .collect();
-
-        let fee_commitment = {
```

Comment:
Previously, the fee was lumped in with the other outputs. Now, the fee is handled separately, which allows it to be included in the balance checks but not in the range proof.